xen-exporter.py

```diff
@@ -62,6 +66,19 @@ def get_or_set(d, key, func, *args):
         d[key] = func(key, *args)
     return d[key]
 
+def collect_poolmaster():
+    xen_user = os.getenv("XEN_USER", "root")
+    xen_password = os.getenv("XEN_PASSWORD", "")
+    xen_host = os.getenv("XEN_HOST", "localhost")
+    verify_ssl = "false"
+    verify_ssl = True if verify_ssl.lower() == "true" else False
+    try:
+       with Xen("https://" + xen_host, xen_user, xen_password, verify_ssl) as xen:
+          poolmaster = xen_host
+    except XenAPI.XenAPI.Failure as e:
+       ipPattern = re.compile('\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}')
```

Comment:
While I'm sure this works, why not just add a new configuration for the poolmaster's address?

Reply:
Copying response from @solipsist01 inline:

> The reason is that poolmaster can change.
what if host 1 is poolmaster, and due to a hardware failure, now another
host is poolmaster.
or, someone is upgrading a xenserver version on all hosts, but let's the
poolmaster on another host.

> then we would need to reconfigure our containers / prometheus setup. This
method makes sure it always works.

Reply:
Thank you for explaining, I think that makes sense to keep this then.